File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0

    const dfs = (idx, sum) => {
        if (idx === numbers.length) {
            if (sum === target) answer++
            return
        }
        dfs(idx + 1, sum + numbers[idx])
        dfs(idx + 1, sum - numbers[idx])
    }
    dfs(0, 0)
    return answer
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0

    def dfs(idx, current_sum):
        nonlocal answer
        if idx == len(numbers):
            if current_sum == target:
                answer += 1
            return

        dfs(idx + 1, current_sum + numbers[idx])
        dfs(idx + 1, current_sum - numbers[idx])

    dfs(0, 0)
    return answer